### PR TITLE
feat: add `Request.setTimeout()` and `BulkLoad.setTimeout()`

### PR DESCRIPTION
**Pull request by chdh**
Implements Request.setTimeout() and BulkLoad.setTimeout(), which can be used to override the timeout specified when the connection was created.

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ module.exports = class Request extends EventEmitter {
   handle: ?any; // TODO: Figure out the type here.
   error: ?Error;
   connection: ?any; // TODO: This should be `Connection`, not `any`.
+  timeout: ?number;
```
Comment (arthurschreiber) [inline src/request.js]: I think the correct type to use here would be:

```js
timeout: number | typeof undefined;
```

`?number` is `number | null | typeof undefined;`, but we shouldn't allow `null` here, but let's not add another incorrect type here.

**Comment by chdh** [comment]
@arthurschreiber 
Thanks for the review. I have completed the changes and adapted the API documentation (#777). Can I merge this PR myself, or will you do that?

**Comment by arthurschreiber** [comment]
Thank you for preparing this! ❤️ 

> Can I merge this PR myself, or will you do that?

I usually try to take on last review pass before merging. In this case, all the changes were looking great, but to ensure that the `fix` commits you made don't show up in the automated changelog generated over at https://github.com/tediousjs/tedious/releases, I squashed all the changes together.

**Comment by chdh** [comment]
@arthurschreiber 
Thanks. (I expected the commits to get squashed.)

**Pull request merged**